tee_logger.go

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package zap
+
+// TeeLogger creates a Logger that duplicates its log calls to two or
+// more loggers. It is similar to io.MultiWriter.
+//
+// The returned logger will initially have the minimum level of all passed
+// loggers. Changing the returned logger's level will change the level on all
+// wrapped loggers.
+//
+// NOTE: Panic and Fatal may not work as expected, since the first core logger
+// will itself Panic or Fatal (interrupting the TeeLogger loop).
+func TeeLogger(logs ...Logger) Logger {
+	switch len(logs) {
+	case 0:
+		return nil
+	case 1:
+		return logs[0]
+	default:
+		lvl := logs[0].Level()
+		for _, log := range logs[1:] {
+			if ll := log.Level(); ll < lvl {
+				lvl = ll
+			}
+		}
+		ml := &multiLogger{
+			Meta: MakeMeta(nil),
```

Comment:
Do we need a meta embedded? I think we're going to be implementing almost all the methods it provides ourselves so not sure if it gives us much.